src/app/utils.py

```diff
@@ -285,3 +291,184 @@ def createLicenseNamespaceIssue(licenseNamespace, token, urlType):
+        data['standardLicenseHeader'] = ''
+    try:
+        if(len(tree.findall('{http://www.spdx.org/license}license/{http://www.spdx.org/license}text')) > 0):
+            textElem = tree.findall('{http://www.spdx.org/license}license/{http://www.spdx.org/license}text')[0]
+            ET.register_namespace('', "http://www.spdx.org/license")
+            textStr = ET.tostring(textElem).strip()
+            if(len(textStr) >= 49 and textStr[:42] == '<text xmlns="http://www.spdx.org/license">' and textStr[-7:] == '</text>'):
+                textStr = textStr[42:]
+                textStr = textStr[:-7].strip().replace('&lt;', '<').replace('&gt;', '>').strip()
+            data['text'] = textStr.strip()
+        else:
+            data['text'] = ''
+    except Exception as e:
+        data['text'] = ''
+    return data
+
+
+def get_xml():
+    my_xml = ET.Element('foo', attrib={'bar': 'bla'})
+    my_str = ET.tostring(my_xml, 'utf-8', short_empty_elements=False)
+    enc = '<?xml version="1.0" encoding="utf-8"?>'
+    enc = enc + my_str.decode('utf-8')
+    return enc
+
+
+def clean(text):
+    """ Clean the XML tags from license text after parsing the XML string.
+    """
+    tags = re.compile('<.*?>')
+    cleanedText = re.sub(tags, '', text)
+    return cleanedText
+
+
+def get_rejected_licenses_issues(urlType):
+    """ Get all the issues that are already rejected by the legal team.
+    """
+    payload = {'state': 'closed', 'labels': 'new license/exception: Not Accepted'}
+    url = TYPE_TO_URL_LICENSE[urlType]
+    res = requests.get(url, params=payload)
+    issues = res.json()
+    return issues
+
+
+def get_yet_not_approved_licenses_issues(urlType):
+    """ Get all the issues that are yet to be approved by the legal team by sorting them with labels.
+    """
+    payload = {'state': 'open', 'labels': 'new license/exception request'}
+    url = TYPE_TO_URL_LICENSE[urlType]
+    response = requests.get(url, params=payload)
+    issues = response.json()
+    newRequestIssues = []
+
+    # Remove issues with Accepted labels
+    for issue in issues:
+        for label in issue.get('labels'):
+            if "new license/exception: Accepted" in label.get('name'):
+                break
+        else:
+            newRequestIssues.append(issue)
+    return newRequestIssues
+
+
+def get_license_data(issues):
+    """ Get license data returns a dictionary with key as license IDs and value as the
+    corresponding license text of these IDs.
+    """
+    licenseIds = []
+    licenseTexts = []
+    for issue in issues:
+        if issue.get('pull_request') is None:
+            licenseInfo = issue.get('body')
+            if '[SPDX-Online-Tools]' in issue.get('title'):
+                licenseIdentifier = re.search(r'(?im)short identifier:\s([a-zA-Z0-9|.|-]+)', licenseInfo).group(1)
+                licenseIds.append(licenseIdentifier)
+                licenseXml = str(LicenseRequest.objects.get(shortIdentifier=licenseIdentifier).xml)
```

Comment:
There should be exception handling here, as its giving me Query does not error .

Reply:
That error is probably due to no issues being labeled as new request or reject ones but good point I will add error handling to it soon. Thanks for pointing that out. :)